core/src/main/scala/kafka/log/LogSegment.scala

```diff
@@ -65,24 +65,18 @@ class LogSegment private[log] (val log: FileRecords,
     val reachedRollMs = timeWaitedForRoll(rollParams.now, rollParams.maxTimestampInMessages) > rollParams.maxSegmentMs - rollJitterMs
     size > rollParams.maxSegmentBytes - rollParams.messagesSize ||
       (size > 0 && reachedRollMs) ||
-      offsetIndex.isFull || timeIndex.isFull || !canConvertToRelativeOffset(rollParams.maxOffsetInMessages)
+      offsetIndex.get.isFull || timeIndex.get.isFull || !canConvertToRelativeOffset(rollParams.maxOffsetInMessages)
   }
 
   def resizeIndexes(size: Int): Unit = {
-    offsetIndex.resize(size)
-    timeIndex.resize(size)
+    offsetIndex.get.resize(size)
+    timeIndex.get.resize(size)
   }
 
-  def sanityCheck(timeIndexFileNewlyCreated: Boolean): Unit = {
-    if (offsetIndex.file.exists) {
-      offsetIndex.sanityCheck()
-      // Resize the time index file to 0 if it is newly created.
-      if (timeIndexFileNewlyCreated)
-        timeIndex.resize(0)
-      timeIndex.sanityCheck()
-      txnIndex.sanityCheck()
-    }
-    else throw new NoSuchFileException(s"Offset index file ${offsetIndex.file.getAbsolutePath} does not exist")
+  def sanityCheck(): Unit = {
+    offsetIndex.get.sanityCheck()
```

Comment:
Should we still make those sanity checks when the indexes are (lazily) created?

Reply:
We don't need to sanity check offse index and the time index anymore.

Reply:
Can you please explain why?

Reply:
Also, let's include the explanation in the PR description.

Reply:
@ijuma The purpose of the sanity check is to detect index corruption (if any) and recover the index. Since in recoverLog() we will recovers every segments after the flushed point in any case,  calling offsetIndex.sanityCheck() and timeIndex.sanityCheck() are redundant operations.